src/gwc/src/main/java/org/geoserver/gwc/layer/DefaultTileLayerCatalog.java

```diff
@@ -172,22 +231,69 @@ public void initialize() {
         LOGGER.info("Loading tile layers from " + baseDir.path());
         Stopwatch sw = Stopwatch.createStarted();
         ExtensionFilter xmlFilter = new Resources.ExtensionFilter("XML");
-        baseDir.list().parallelStream().filter(r -> xmlFilter.accept(r)).forEach(this::load);
+        // do not thrash the filesystem if there are several cores by using the common
+        // pool
+        ForkJoinPool pool =
+                new ForkJoinPool(
+                        INITIALIZATION_PARALLELISM, INITIALLIZATION_THREAD_FACTORY, null, false);
+        try {
+            pool.submit(
+                            () ->
+                                    baseDir.list()
+                                            .parallelStream()
+                                            .filter(r -> xmlFilter.accept(r))
+                                            .forEach(this::initializationLoad))
+                    .join();
+        } finally {
+            pool.shutdownNow();
+        }
         LOGGER.info(String.format("Loaded %,d tile layers in %s", layersById.size(), sw.stop()));
         this.initialized = true;
     }
 
+    /**
+     * Called during initialization, inside a forkjoinpool thread. Uses {@link
+     * #INITIALIZATION_SERIALIZER} ThreadLocal safely, it'll be cleaned as the threads in the
+     * forkjoinpool die
+     */
+    private GeoServerTileLayerInfoImpl initializationLoad(Resource res) {
+        XStream unmarshaller = INITIALIZATION_SERIALIZER.get();
+        if (unmarshaller == null) {
+            unmarshaller = newXStream();
+            INITIALIZATION_SERIALIZER.set(unmarshaller);
+        }
+        GeoServerTileLayerInfoImpl info;
+        try {
+            info = depersist(res, unmarshaller);
+        } catch (Exception e) {
+            LOGGER.log(
+                    Level.SEVERE,
+                    "Error depersisting tile layer information from file " + res.name(),
+                    e);
+            return null;
+        }
+        saveInternal(info);
+        if (LOGGER.isLoggable(Level.FINER)) {
+            LOGGER.finer("Loaded tile layer '" + info.getName() + "'");
+        }
+        return info;
+    }
+
     @Override
     public GeoServerTileLayerInfo getLayerById(final String id) {
         checkInitialized();
         GeoServerTileLayerInfo layer = layersById.get(id);
         return layer == null ? null : layer.clone();
     }
 
-    private synchronized void checkInitialized() {
-        if (!initialized) {
-            initialize();
-        }
+    /**
+     * Precondition check all public methods should make before proceeding to ensure they've been
+     * called on an initialized state
+     *
+     * @throws IllegalStateException if this layer catalog has not been initialized yet
+     */
+    private void checkInitialized() {
```

Comment:
Hmm... this method changed from "check if initialized and if not initialize" to "check if initialized and throw an exception if not". So I'm guessing that's why there is an explicit call to initialize elsewhere. Not a big deal, just wondering.

Reply:
Correct. DefaultTileLayerCatalog initialization is and shall be managed by the lifecycle management methods (reset() and initialize()). `checkInitialized()` repurposed as public methods precondition check.